product/roundhouse.console/roundhouse.console.csproj

```diff
@@ -1,65 +1,44 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
   <PropertyGroup>
+    <TargetFramework>net461</TargetFramework>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProductVersion>9.0.30729</ProductVersion>
-    <SchemaVersion>2.0</SchemaVersion>
-    <ProjectGuid>{97CA60FD-D550-4098-8B59-E6CD1F1A3AF2}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <OutputPath>bin\</OutputPath>
     <RootNamespace>roundhouse.console</RootNamespace>
     <AssemblyName>rh</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
-    <ApplicationIcon>..\..\docs\logo\roundhouse.ico</ApplicationIcon>
-    <IsWebBootstrapper>false</IsWebBootstrapper>
-    <FileUpgradeFlags>
-    </FileUpgradeFlags>
-    <OldToolsVersion>3.5</OldToolsVersion>
-    <UpgradeBackupLocation />
-    <PublishUrl>publish\</PublishUrl>
-    <Install>true</Install>
-    <InstallFrom>Disk</InstallFrom>
-    <UpdateEnabled>false</UpdateEnabled>
-    <UpdateMode>Foreground</UpdateMode>
-    <UpdateInterval>7</UpdateInterval>
-    <UpdateIntervalUnits>Days</UpdateIntervalUnits>
-    <UpdatePeriodically>false</UpdatePeriodically>
-    <UpdateRequired>false</UpdateRequired>
-    <MapFileExtensions>true</MapFileExtensions>
-    <ApplicationRevision>0</ApplicationRevision>
-    <ApplicationVersion>1.0.0.%2a</ApplicationVersion>
-    <UseApplicationTrust>false</UseApplicationTrust>
-    <BootstrapperEnabled>true</BootstrapperEnabled>
-    <FileUpgradeFlags>
-    </FileUpgradeFlags>
-    <OldToolsVersion>3.5</OldToolsVersion>
-    <UpgradeBackupLocation />
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
-    <RestorePackages>true</RestorePackages>
-    <TargetFrameworkProfile />
+    <NoWarn>NU1701</NoWarn>
+    <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
+
+    <VersionPrefix>0.8.9</VersionPrefix>
+    <VersionSuffix>alpha</VersionSuffix>

```

Comment:
How do these get updated?

Reply:
They don't. I override them on command-line with MSBuild properties when packing (in build.ps1). I generate semantic version using GitVersion.exe (probably same logic as in the nant build files) . Maybe just leave them as a "default" or easy-to-understand-that-is-dummy value in the .csproj?

Do you have any thoughts on external dependencies, such as GitVersion.exe? Should we check in the binaries, or should we expect people to use chocolatey to install themselves?

Reply:
GitVersion is awesome.

Reply:
There is a concept of setup, where all the tools are brought to a repository. That setup should give a developer everything they need to do build.

Reply:
True, true. We could make a `setup.ps1` which pulls chocolatey from the web, etc, and uses that to install GitVersion. but maybe a bit intrusive. Is it possible to install choco packages without admin? If not, this will definitely not work for build servers, etc. I love having the same script on build servers as local, so that would be nice, to create a setup.ps1, which assures all necessary tools are in place, and then the build.ps1 (or cake, maybe, preferably), could do the build part itself. 

nuget it part of the build chain, but the .exe of chocolatey is not in the nuget package (which makes sense).

Reply:
The good news? Cake does all of this already.